test/integration/nighthawk_grpc_service.py

```diff
@@ -0,0 +1,71 @@
+import json
+import logging
+import os
+import socket
+import subprocess
+import sys
+import tempfile
+import threading
+import time
+from string import Template
+
+from common import IpVersion
+
+
+class NighthawkGrpcService(object):
+  """
+    Base class for running a server in a separate process.
+    """
+
+  def __init__(self, server_binary_path, server_ip, ip_version):
+    assert ip_version != IpVersion.UNKNOWN
+    self.ip_version = ip_version
+    self.server_binary_path = server_binary_path
+    self.server_thread = threading.Thread(target=self.serverThreadRunner)
+    self.server_process = None
+    self.server_ip = server_ip
+    self.socket_type = socket.AF_INET6 if ip_version == IpVersion.IPV6 else socket.AF_INET
+    self.server_port = 0
+    self.address_file = ""
+
+  def serverThreadRunner(self):
```

Comment:
Similarly for methods - can we indicate what the public API is by marking anything private with a leading underscore?

def _serverThreadRunner(self):